mock_service/doc/concept.md

```diff
@@ -0,0 +1,198 @@
+# Concept
+## Introduction
+
+As VApp developer I want to have a mock environment where all VSS data points are available and in which actuators will respond when they are requested to change their value.
+
+**Ideally, I should not care about data point availability during development time.**
+
+## Scope
+
+The scope of the Vehicle Mock is a CLI application without a GUI and without phyiscal simulation. It shall be usable to
+
+* configure the mock with expected actuator and sensor behavior to develop an application against
+* configure the mock with actions and reactions to use for integration tests
+
+## Functional requirements
+
+| id | version | description |
+|:---|:-------:|:-----------|
+| #F1  | MVP     | Vehicle Mock shall connect to *Kuksa Data Broker* via gRPC and act as a data provider.
+| #F2 | MVP | Vehicle Mock shall be able to accept external configuration for each data point that it is supposed to mock (= mocked data point).
+| #F3 | MVP | Mocked data points shall be programmable with predefined behaviors via configuration.
+| #F4 | MVP | Mocked data points shall be initializeable with a predefined value via configuration.
+| #F5 | MVP | A single mocked data point shall be configurable by its exact path (i.e. `Vehicle.Cabin.Seat.Row1.Pos1.Position`)
+| #F6 | post-MVP | A collection of mocked data points shall be configurable by specifing the wild card `*` at a specific sub-branch (i.e. `Vehicle.Cabin.Seat.*`)
+| #F7 | MVP | Vehicle Mock shall be able to execute predefined behaviors once a certain condition is met ...
+| #F7.1 | MVP | ... a condition evaluates to true
+| #F7.2 | MVP | ... an event is received from *Kuksa Data Broker*
+| #F8 | MVP | Vehicle Mock shall be able to respond to `actuator_target` requests for all mocked data points, even those without a behavior
+| #F8.1 | MVP | if no behavior is available the `actuator_target` request shall be responded to with an immediate setting of the value
```

Comment:
Does this mean that it should be possible to define a mocked actuator without specifying any behaviour, i.e.
```python
mock_datapoint(
    path="Vehicle.Cabin.HVAC.IsFrontDefrosterActive",
    initial_value=False,
)
```
with the expectation that it will set the value to the actuator target.

Because that doesn't seem to be possible at the moment (as behaviour is required argument).

Reply:
You are right, I added a list as default value to behaviors to make behavious optional.

Reply:
Using this doesn't generate an error when starting the mock service now:

```python
mock_datapoint(
    path="Vehicle.Cabin.HVAC.IsFrontDefrosterActive",
    initial_value=False,
)
```

But as far as I can tell, setting an actuator target doesn't result in the value being set by the mock.

In other words, this requirement isn't being fulfilled.
> if no behavior is available the `actuator_target` request shall be responded to with an immediate setting of the value